addons/sale_timesheet/models/hr_timesheet.py

```diff
@@ -156,6 +157,21 @@ def _get_timesheets_to_merge(self):
         res = super()._get_timesheets_to_merge()
         return res.filtered(lambda l: not l.timesheet_invoice_id or l.timesheet_invoice_id.state != 'posted')
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if vals.get('so_line'):
```

Comment:
To avoid to many nested if and stuffs:
```suggestion
            if not vals.get('so_line'):
                continue
```